server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/monetaryunit/MonetaryUnitExtension.java

```diff
@@ -0,0 +1,124 @@
+import com.generalbytes.batm.common.currencies.FiatCurrency;
+import com.generalbytes.batm.server.extensions.*;
+import com.generalbytes.batm.server.extensions.FixPriceRateSource;
+import com.generalbytes.batm.server.extensions.extra.dash.sources.coinmarketcap.CoinmarketcapRateSource;
+import com.generalbytes.batm.server.extensions.extra.bitcoin.sources.coinpaprika.CoinPaprikaRateSource;
+import com.generalbytes.batm.server.extensions.extra.bitcoin.sources.coingecko.CoinGeckoRateSource;
+import com.generalbytes.batm.server.extensions.extra.monetaryunit.wallets.monetaryunitd.MonetaryUnitRPCWallet;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.StringTokenizer;
+
+public class MonetaryUnitExtension extends AbstractExtension{
+    @Override
+    public String getName() {
+        return "BATM MonetaryUnit extension";
+    }
+
+    @Override
+    public IWallet createWallet(String walletLogin) {
+        if (walletLogin !=null && !walletLogin.trim().isEmpty()) {
+            StringTokenizer st = new StringTokenizer(walletLogin,":");
+            String walletType = st.nextToken();
+
+            if ("monetaryunitd".equalsIgnoreCase(walletType)) {
+                //"monetaryunitd:protocol:user:password:ip:port:accountname"
+
+                String protocol = st.nextToken();
+                String username = st.nextToken();
+                String password = st.nextToken();
+                String hostname = st.nextToken();
+                String port = st.nextToken();
+                String accountName ="";
+                if (st.hasMoreTokens()) {
+                    accountName = st.nextToken();
+                }
+
+
+                if (protocol != null && username != null && password != null && hostname !=null && port != null && accountName != null) {
+                    String rpcURL = protocol +"://" + username +":" + password + "@" + hostname +":" + port;
+                    return new MonetaryUnitRPCWallet(rpcURL,accountName);
+                }
+            }
+            if ("monetaryunitdemo".equalsIgnoreCase(walletType)) {
+
+                String fiatCurrency = st.nextToken();
+                String walletAddress = "";
+                if (st.hasMoreTokens()) {
+                    walletAddress = st.nextToken();
+                }
+
+                if (fiatCurrency != null && walletAddress != null) {
+                    return new DummyExchangeAndWalletAndSource(fiatCurrency, CryptoCurrency.MUE.getCode(), walletAddress);
+                }
+            }
+        }
+        return null;
+    }
+    
+    @Override
+    public IRateSource createRateSource(String sourceLogin) {
+        if (sourceLogin != null && !sourceLogin.trim().isEmpty()) {
+            StringTokenizer st = new StringTokenizer(sourceLogin,":");
+            String rsType = st.nextToken();
+            
+             if ("coinmarketcap".equalsIgnoreCase(rsType)) {
```

Comment:
not really necessary as the coinmarketcap rate source with MUE support can be instantiated from the original extension it came with, but the code does no harm. I can merge it even like this.